pkg/operators/kubenameresolver/kubenameresolver.go

```diff
@@ -134,6 +138,176 @@ func (m *KubeNameResolverInstance) EnrichEvent(ev any) error {
+
+		k8sField := ds.GetField("k8s")
+		if k8sField == nil {
+			logger.Debugf("> no k8s fields found")
+			continue
+		}
+
+		k8sAccesors := k8sAccesors{}
+		var err error
+
+		k8sAccesors.PodName = ds.GetField("k8s.podName")
+		if k8sAccesors.PodName == nil {
+			logger.Warnf("No podName field found in datasource %q", ds.Name())
+			continue
+		}
+
+		k8sAccesors.Namespace = ds.GetField("k8s.namespace")
+		if k8sAccesors.Namespace == nil {
+			logger.Warnf("No namespace field found in datasource %q", ds.Name())
+			continue
+		}
+
+		k8sAccesors.Owner = ds.GetField("k8s.owner")
+		if k8sAccesors.Owner == nil {
+			logger.Warnf("No owner field found in datasource %q", ds.Name())
+			continue
+		}
+
+		k8sAccesors.PodLabels = ds.GetField("k8s.podLabels")
+		if k8sAccesors.PodLabels == nil {
+			logger.Warnf("No podLabels field found in datasource %q", ds.Name())
+			continue
+		}
+
+		// Create 2 new fields
+		k8sAccesors.HostIP, err = k8sField.AddSubField("hostIP", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+		if err != nil {
+			return nil, fmt.Errorf("adding field %q: %w", "hostIP", err)
+		}
+		k8sAccesors.PodIP, err = k8sField.AddSubField("podIP", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+		if err != nil {
+			return nil, fmt.Errorf("adding field %q: %w", "podIP", err)
+		}
+
+		logger.Debugf("> Found fields for DS %q", ds.Name())
+		accessors[ds] = k8sAccesors
+	}
+
+	// No endpoints found, nothing to do
+	if len(accessors) == 0 {
+		return nil, nil
+	}
+
+	k8sInventory, err := common.GetK8sInventoryCache()
+	if err != nil {
+		return nil, fmt.Errorf("creating k8s inventory cache: %w", err)
+	}
+
+	return &KubeNameResolverInstance{
+		k8sInventory: k8sInventory,
+		accessors:    accessors,
+	}, nil
+}
+
+func (m *KubeNameResolverInstance) enrichSingle(data datasource.Data, accessor k8sAccesors) error {
+	podName, err := accessor.PodName.String(data)
+	if err != nil {
+		return fmt.Errorf("getting podName: %w", err)
+	}
+	namespace, err := accessor.Namespace.String(data)
+	if err != nil {
+		return fmt.Errorf("getting namespace: %w", err)
+	}
+
+	pod := m.k8sInventory.GetPodByName(namespace, podName)
+	if pod == nil {
+		return nil
+	}
+
+	accessor.HostIP.PutString(data, pod.Status.HostIP)
+	accessor.PodIP.PutString(data, pod.Status.PodIP)
+
+	// When the pod belongs to Deployment, ReplicaSet or DaemonSet, find the
+	// shorter name without the random suffix. That will be used to
+	// generate the network policy name.
+	if pod.OwnerReferences != nil {
+		// Trim the pod name to remove the suffix after the second to last dash
+		// (e.g. "pod-name-1234567890-abcde" -> "pod-name")
+		numDashes := 0
+		lastIndex := len(pod.Name) - 1
+		for ; lastIndex >= 0; lastIndex-- {
+			if pod.Name[lastIndex] != '-' {
+				continue
+			}
+
+			numDashes++
+			if numDashes == 2 {
+				break
+			}
+		}
+
+		if numDashes == 2 {
+			accessor.Owner.PutString(data, pod.Name[:lastIndex])
+		}
+	}
```

Comment:
Shouldn't this be part of the gadget itself and not of this operator?

Reply:
Makes sense, this is something too specific (Also the comment :D)